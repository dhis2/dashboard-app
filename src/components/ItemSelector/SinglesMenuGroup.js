import React, { Fragment } from 'react';
import { connect } from 'react-redux';

import HeaderMenuItem from './HeaderMenuItem';
import ContentMenuItem from './ContentMenuItem';
import { acAddDashboardItem } from '../../actions/editDashboard';

export const SinglesMenuGroup = ({ acAddDashboardItem, category }) => {
    const addToDashboard = ({ type, content }) => () => {
        acAddDashboardItem({ type, content });
    };

    return (
        <Fragment>
            <HeaderMenuItem title={category.header} />
            {category.items.map(item => (
                <ContentMenuItem
                    key={item.type}
                    type={item.type}
                    name={item.name}
                    onInsert={addToDashboard(item)}
                />
            ))}
        </Fragment>
    );
};

export default connect(
    null,
    { acAddDashboardItem }
)(SinglesMenuGroup);
