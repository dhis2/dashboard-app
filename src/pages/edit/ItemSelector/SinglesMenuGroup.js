import PropTypes from 'prop-types'
import React from 'react'
import { connect } from 'react-redux'
import { tSetDashboardItems } from '../../../actions/editDashboard'
import ContentMenuItem from './ContentMenuItem'
import HeaderMenuItem from './HeaderMenuItem'

const SinglesMenuGroup = ({ onAddItem, category }) => {
    const addToDashboard =
        ({ type, content }) =>
        () => {
            onAddItem({ type, content })
        }

    return (
        <>
            <HeaderMenuItem title={category.header} />
            {category.items.map(item => (
                <ContentMenuItem
                    key={item.type}
                    type={item.type}
                    name={item.name}
                    onInsert={addToDashboard(item)}
                />
            ))}
        </>
    )
}

SinglesMenuGroup.propTypes = {
    category: PropTypes.object,
    onAddItem: PropTypes.func,
}

export default connect(null, {
    onAddItem: item => dispatch => {
        dispatch(tSetDashboardItems(item))
    },
})(SinglesMenuGroup)
