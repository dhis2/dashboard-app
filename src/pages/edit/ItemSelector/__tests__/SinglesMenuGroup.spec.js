import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import SinglesMenuGroup from '../SinglesMenuGroup.js'

const mockStore = configureMockStore()

test('renders SingleMenuGroup', () => {
    const store = {}

    const props = {
        category: {
            header: 'ponies',
            items: [
                {
                    type: 'colorful',
                    name: 'Rainbow Dash',
                },
                {
                    type: 'greytone',
                    name: 'Twilight',
                },
            ],
        },
    }

    const { container } = render(
        <Provider store={mockStore(store)}>
            <SinglesMenuGroup {...props} />
        </Provider>
    )
    expect(container).toMatchSnapshot()
})
