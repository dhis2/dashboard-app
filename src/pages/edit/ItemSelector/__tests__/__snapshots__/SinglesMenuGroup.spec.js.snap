// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SingleMenuGroup 1`] = `
<div>
  <li
    class="jsx-3508410433 item disabled dense"
    data-test="dhis2-uicore-menuitem"
  >
    <a
      class="jsx-3508410433"
    >
      <span
        class="jsx-3508410433 label"
      >
        <span
          style="color: rgb(64, 75, 90); font-weight: 600;"
        >
          ponies
        </span>
      </span>
    </a>
  </li>
  <li
    class="jsx-3508410433 "
    data-test="menu-item-Rainbow Dash"
  >
    <a
      class="jsx-3508410433"
    >
      <span
        class="jsx-3508410433 icon"
      >
        <svg
          color="#6C7787"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm0 12a1 1 0 110 2 1 1 0 010-2zm0-10a4 4 0 011.155 7.83l-.155.043V15h-2v-3h1a2 2 0 10-1.995-2.15L10 10H8a4 4 0 014-4z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="jsx-3508410433 label"
      >
        <div
          class="menuItem"
        >
          <div
            class="label"
          >
            <span>
              Rainbow Dash
            </span>
          </div>
          <button
            class="buttonInsert"
          >
            Insert
          </button>
        </div>
      </span>
    </a>
  </li>
  <li
    class="jsx-3508410433 "
    data-test="menu-item-Twilight"
  >
    <a
      class="jsx-3508410433"
    >
      <span
        class="jsx-3508410433 icon"
      >
        <svg
          color="#6C7787"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm0 12a1 1 0 110 2 1 1 0 010-2zm0-10a4 4 0 011.155 7.83l-.155.043V15h-2v-3h1a2 2 0 10-1.995-2.15L10 10H8a4 4 0 014-4z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="jsx-3508410433 label"
      >
        <div
          class="menuItem"
        >
          <div
            class="label"
          >
            <span>
              Twilight
            </span>
          </div>
          <button
            class="buttonInsert"
          >
            Insert
          </button>
        </div>
      </span>
    </a>
  </li>
</div>
`;
